### DISCUSSION TITLE
Missing statement about the needed hugo version

### DISCUSSION THREAD

**Discussion by irkode** [hugo-sid/hugo-blog-awesome#249] [General]
put that here cause I couln't make the choice for bug/feature/...

according to the README `hugo extended` is needed and in some of the files there's a min version of *0.87.0*

but the _exampleSite_ fails fe. with `0.92.2` . Theme has recent releases, I suppose the min requirement is much higher these days and should be
* documented somewhere
   ReleaseNotes, README, Setup
* maybe code to catch non matching versions
   hugo.toml to the theme so it will warn (at least for module usage)
   maybe a check in baseof...

**Comment by hugo-sid** [comment]
Thanks @irkode for sharing the idea. We can set a minimum version requirement so that the build will fail if the version of Hugo is anything less than that. But we have to figure out what's the minimum version. Or else, I can simply set the present latest version as the minimum version.

**Comment by irkode** [comment]
Just some thoughts:

I've seen so many people keeping older versions of hugo with their running blogs. If you raise too aggressive you will exclude all from applying the next release of your theme.

Keep in mind that they may have own templates or customized parts. So you can _never_ be sure their code will run with the newest.

There have been some breaking changes over the time. For an example see: [Hugo Forum Topic 52209](https://discourse.gohugo.io/t/markdown-changes-v131-136-5/52209) where a change in 134 compiles fine, but breaks the sites display only ... and there are others with deprecations and removed stuff that completely fail

I would:
- figure our which old version compiles your exampleSite and looks fine
- use that as minimal version for your v1
- for the last mile: figure out which next version of hugo breaks your theme and set this as max version

- raise hugo to a recent one with a v2

So you support running blogs with older hugos and have an update path for users users

Ofc will add some more dependency and checks on your side. So you have to find a path that you can support...

That said

You already do a good job with keeping your theme up to date (workflow/test deploy/visual check...)

I have not seen many themes around that compile with latest hugo versions _without_ any warnings

Good work
